pkg/controller/pod.go

```diff
@@ -248,6 +248,20 @@ func (pnc *PodNetworksController) processNextWorkItem() bool {
 		}
 	}
 
+	// handleDynamicInterfaceRequest modifies the pod annotation with the latest status
+	// so the updated status annotation can be obtained from there.
+	newIfaceStatus, err := annotations.PodDynamicNetworkStatus(pod)
+	if err != nil {
+		klog.Errorf("failed while getting network status annotation: %v", err)
+		return true
+	}
+
+	// apply the network status using the Kubernetes API.
+	if err := nadutils.SetNetworkStatus(pnc.k8sClientSet, pod, newIfaceStatus); err != nil {
+		klog.Errorf("failed while setting network status annotation: %v", err)
+		return true
+	}
```

Comment:
The commit message says:
> The controller is now setting the pod network status annotation only
once in the reconcile loop to avoid conflicts when a network attachment
is added and another one removed at the same time.

But I don't think that's true; IIUC, you're setting it 3 times - once in `addNetworks`, once in `remove` networks, and a final one (this one) at the end of `processNextWorkItem` . 

Right ?

Reply:
Add and remove is only updating the pod object, it doesn't update the pod in Kubernetes